app/src/main/java/com/bignerdranch/android/criminalintent/CrimeListFragment.kt

```diff
@@ -9,28 +9,26 @@ import android.widget.ImageView
 import android.widget.TextView
 import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModelProvider
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import java.text.SimpleDateFormat
 import java.util.*
+import androidx.lifecycle.Observer
 
 private const val TAG = "CrimeListFragment"
 
 class CrimeListFragment: Fragment() {
 
     private lateinit var crimeRecyclerView: RecyclerView
-    private var adapter: CrimeAdapter? = null
+    private var adapter: CrimeAdapter? = CrimeAdapter(emptyList())
```

Comment:
Why do you need the `adapter` to be a nullable type, if you instantiate it directly and never nullify it? Wouldn't it be better to write
```
private val adapter: CrimeAdapter = CrimeAdapter(emptyList())
```